assets/javascripts/discourse/components/chat-channel-delete-modal-inner.js

```diff
@@ -0,0 +1,55 @@
+import Component from "@ember/component";
+import { isTesting } from "discourse-common/config/environment";
+import { later } from "@ember/runloop";
+import { isEmpty } from "@ember/utils";
+import I18n from "I18n";
+import discourseComputed from "discourse-common/utils/decorators";
+import { action } from "@ember/object";
+import { ajax } from "discourse/lib/ajax";
+import { inject as service } from "@ember/service";
+import { popupAjaxError } from "discourse/lib/ajax-error";
+
+export default Component.extend({
+  chat: service(),
+  tagName: "",
+  chatChannel: null,
+  channelNameConfirmation: null,
+
+  @discourseComputed("deleting", "channelNameConfirmation")
+  buttonDisabled(deleting, channelNameConfirmation) {
+    if (deleting) {
+      return true;
+    }
+
+    if (
+      isEmpty(channelNameConfirmation) ||
+      channelNameConfirmation.toLowerCase() !==
+        this.chatChannel.title.toLowerCase()
+    ) {
+      return true;
+    }
+    return false;
+  },
+
+  @action
+  deleteChannel() {
+    this.set("deleting", true);
+    return ajax(`/chat/chat_channels/${this.chatChannel.id}.json`, {
+      method: "DELETE",
+      data: { channel_name_confirmation: this.channelNameConfirmation },
+    })
+      .then(() => {
+        this.appEvents.trigger("modal-body:flash", {
+          text: I18n.t("chat.channel_delete.process_started"),
+          messageClass: "success",
+        });
+        later(() => {
+          if (!isTesting()) {
+            window.location.reload();
+          }
+        }, 3000);
+      })
```

Comment:
I didn't pull down the changes locally but I think having the page suddenly reload might be surprising especially if I've navigated away from the page. Is the reload to used to clear the channel object from chat?

Reply:
Yes, and I've already used the same pattern in the Archive modal. You get shown a message before this happens so it shouldn't be too surprising.

Maybe there is some way to cancel this when the modal closes?

Reply:
Hmm how about just displaying a different view for the chat channel once the delete request is successful? Deferring a page reload based on an arbitrary duration feels like it will be hard to get right UX wise. I might be reading through the message in the modal midway when the page is refreshed or I might have closed the modal.

Reply:
Fair enough will think of something else for this and the archive modal, thinking closer about it the refresh isn't actually necessary. The user is on the /chat/browse page when they do this action.